contracts/Core/BlockManager.sol

```diff
@@ -65,68 +57,88 @@ contract BlockManager is Initializable, ACL, BlockStorage {
     }
 
     //anyone can give sorted votes in batches in dispute state
     function giveSorted(
-        uint256 epoch,
-        uint256 assetId,
+        uint32 epoch,
+        uint8 assetId,
         uint256[] memory sorted
     ) external initialized checkEpoch(epoch) checkState(parameters.dispute()) {
-        uint256 medianWeight = voteManager.getTotalInfluenceRevealed(epoch, assetId) / (2);
+        uint256 medianWeight = voteManager.getTotalInfluenceRevealed(epoch) / (2);
         uint256 accWeight = disputes[epoch][msg.sender].accWeight;
         uint256 lastVisited = disputes[epoch][msg.sender].lastVisited;
         if (disputes[epoch][msg.sender].accWeight == 0) {
             disputes[epoch][msg.sender].assetId = assetId;
         } else {
             require(disputes[epoch][msg.sender].assetId == assetId, "AssetId not matching");
+            // require(disputes[epoch][msg.sender].median == 0, "median already found");
         }
         for (uint256 i = 0; i < sorted.length; i++) {
             require(sorted[i] > lastVisited, "sorted[i] is not greater than lastVisited");
             lastVisited = sorted[i];
-            accWeight = accWeight + (voteManager.getVoteWeight(epoch, assetId, sorted[i]));
+            accWeight = accWeight + (voteManager.getVoteWeights(epoch, assetId, sorted[i]));
             if (disputes[epoch][msg.sender].median == 0 && accWeight > medianWeight) {
                 disputes[epoch][msg.sender].median = sorted[i];
+                // break;
             }
         }
         disputes[epoch][msg.sender].lastVisited = lastVisited;
         disputes[epoch][msg.sender].accWeight = accWeight;
     }
 
     // //if any mistake made during giveSorted, resetDispute and start again
-    function resetDispute(uint256 epoch) external initialized checkEpoch(epoch) checkState(parameters.dispute()) {
+    function resetDispute(uint32 epoch) external initialized checkEpoch(epoch) checkState(parameters.dispute()) {
         disputes[epoch][msg.sender] = Structs.Dispute(0, 0, 0, 0);
     }
 
-    function getBlock(uint256 epoch) external view returns (Structs.Block memory _block) {
+    function confirmBlock(uint32 epoch) external initialized onlyRole(parameters.getBlockConfirmerHash()) {
+        for (uint8 i = 0; i < sortedProposedBlockIds[epoch - 1].length; i++) {
+            uint8 blockId = sortedProposedBlockIds[epoch - 1][i];
+            if (proposedBlocks[epoch - 1][blockId].valid) {
+                blocks[epoch - 1] = proposedBlocks[epoch - 1][blockId];
+                uint32 proposerId = proposedBlocks[epoch - 1][blockId].proposerId;
+
+                emit BlockConfirmed(epoch - 1, proposerId, proposedBlocks[epoch - 1][i].medians, block.timestamp);
+                rewardManager.giveBlockReward(proposerId, epoch);
+                return;
+                // for (uint8 j = 0; j < proposedBlocks[epoch - 1][i].medians.length; j++) {
```

Comment:
Since this has been commented out, i think we should comment out or remove fulfillAsset from asset manager as that function is not been called by any contract anymore

Reply:
removed